BatterySensor.h

```diff
@@ -365,21 +375,34 @@ class IrqInternalBatt {
     ADCSRA &= ~((1 << ADIE) | (1 << ADIF));  // disable interrupt
     while (ADCSRA & (1 << ADSC)) ;  // wait finish
     irq();    // ensure value is read
+    if (m_ActivationPin > 0)
+      pinMode(m_ActivationPin, INPUT);
   }
   /**
    * Enable the continues measurement of the battery voltage
    * Called by HAL after return from idle/sleep state
    * Call this after the application doesn't need ADC longer
    */
+
   void unsetIdle () {
     //DDECLN(__gb_BatCurrent);
+    if (m_ActivationPin > 0) {
+      pinMode(m_ActivationPin, OUTPUT);
+      digitalWrite(m_ActivationPin, LOW);
+    }
     ATOMIC_BLOCK( ATOMIC_RESTORESTATE ) {
       __gb_BatCount = 0; // reset irq counter
       __gb_BatIrq = irq; // set irq method
     }
     ADMUX &= ~(ADMUX_REFMASK | ADMUX_ADCMASK);
-    ADMUX |= ADMUX_REF_AVCC;      // select AVCC as reference
-    ADMUX |= ADMUX_ADC_VBG;       // measure bandgap reference voltage
+    if (m_SensPin > 0) {
+      _delay_ms(5);
```

Comment:
Der delay gefällt mir gar nicht. Damit wird das Aufwachen grundsätzlich verzögert. Das kann dann wieder zu Problemen bei den Sendeslots führen.

Reply:
Damit geht es zumindest zuverlässig, nachdem der ADC von der LowPower deaktiviert wurde.
Eine andere Möglichkeit, diese "Einschwingphase" nach dem Wecken aus dem Sleep zu überbrücken, habe ich nicht gefunden.

Ich werde dann später noch mal schauen, wie hoch der Stromverbrauch im Sleep mit aktiviertem ADC ist.

Reply:
> Das kann dann wieder zu Problemen bei den Sendeslots führen. 

Das betrifft doch eigentlich nur die Kopplung von TH-Sensoren mit Thermostaten?
Dann ist nur für diesen einen Fall die externe Interrupt-Messung (sofern überhaupt jmd. benötigen sollte) ungeeignet. 😎

Reply:
Man könnte uach die batSkip behandlung in die unsetIdle verlegen. Also wenn X mal kein Wert übernommen wurde, geht man nach Aktivierung des ADC in eine Schleife und wartet, dass die gemessenen Werte stabil werden - oder bricht nach max Durchläufen ab. Dann wird es halt nicht jedes mal verzögert und auch nur so lange wie nötig.